### DISCUSSION TITLE
Unable to Flash -- ESP Flashing Permissions in Fedora -- Resolved

### DISCUSSION THREAD

**Discussion by ggalt** [toverainc/willow#294] [General]
This is just a quick note to save some people time.  At least for me, my Fedora 38 system (admittedly, one that has been around since FC1) did not put my user into the "dialout" group.  Since my tty devices all are owned by root, but are in the dialout group, I initially could not flash using the web interface (Chrome is running with my permissions).  Adding myself to the dialout group fixed the problem, but it took a little poking around to find the issue.  I got the needed hint here: https://github.com/toverainc/willow/issues/60#issuecomment-1569260817.  I hope this helps somone else.  Thanks!

**Comment by kristiankielhofner** [comment]
Thanks for this!

Linux permissions are something that trip up a lot of users... For many distros (but not all!) you need to add yourself to the dialout group (as you've noted) and then logout and log back in, which is less than ideal. However, there really isn't anything we can do about it other than document it. Of course on Windows and Mac it always "just works". IMO not adding regular users to the dialout (or similar) group by default on Linux is terrible UX and a hold-over from the Linux/UNIX (decades ago) days when hosts were sometimes multi-user servers for serial terminals.

We want to update the docs but with the variety of Linux distros and varying skill level of users we're still kind of figuring out how to best go about this - especially trying to come up with some mechanism in the web flasher itself that says something like "We aren't able to write to that port, check the docs here", etc.

Depending on the skill level of the user there have already been plenty of scenarios where we advise them to just flash from a Windows or Mac machine they have around, which is of course less than ideal but as I've highlighted the Linux desktop situation is a bit of a mess when it comes to this.
